python/src/trezorlib/firmware/sanity_struct.py

```
@@ -0,0 +1,136 @@
+        cls: t.Type[Self], data: bytes, *, strict: bool = STRICT_SANITY_CHECK_DEFAULT
+    ) -> Self:
+        parsed_image = super().parse(data)
+        try:
+            parsed_image.sanity_check(data)
+        except SanityCheckError as e:
+            if strict:
+                raise
+            LOG.warning(e.get_warning_message())
+        return parsed_image
+
+    def sanity_check(self, image: bytes, errors: list[str] = []) -> None:
```

Ruff (0.15.13)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)